Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp

```diff
@@ -224,7 +224,13 @@ void WebFullScreenManager::enterFullScreenForElement(WebCore::Element* element)
     if (m_mainVideoElement)
         videoDimensions = FloatSize(m_mainVideoElement->videoWidth(), m_mainVideoElement->videoHeight());
 #endif
-    m_page->injectedBundleFullScreenClient().enterFullScreenForElement(m_page.ptr(), element, m_element->document().quirks().blocksReturnToFullscreenFromPictureInPictureQuirk(), isVideoElement, videoDimensions);
+    m_page->injectedBundleFullScreenClient().enterFullScreenForElement(m_page.ptr(), element, m_element->document().quirks().blocksReturnToFullscreenFromPictureInPictureQuirk(), isVideoElement, videoDimensions, mode);
+
+    if (mode == WebCore::HTMLMediaElementEnums::VideoFullscreenModeInWindow) {
+        willEnterFullScreen(mode);
+        didEnterFullScreen();
+        m_isInInWindowFullScreenMode = true;
```

Comment:
“isInIn” kinda sounds a bit weird

Reply:
I definitely agree haha. I'll try to think of something better

Reply:
I changed it to m_inWindowFullScreenMode